### PR TITLE
Fixed CarouselView Loop="False" rendering items incorrectly

### LINKED ISSUE #25192: CarouselView Loop="False" renders incorrectly items ( net8, net9 rc2)
### Description

Using a CarouselView in a .net maui project ( tested with v8.0.91 & .NET9 RC2) with` Loop="False" `results in an incorrect render. If we set` Loop="True"` the items are rendered correctly.

**With` Loop=True`**
Imgur Image

**Using `Loop=False`**

Imgur Image

It happens in 8.0.92 and in .net 9 rc2
The code can be found here: https://github.com/bcaceiro/net9-repros/tree/main/CarouselBug

### Steps to Reproduce

1. Clone https://github.com/bcaceiro/net9-repros/ -> CarouselBug solution
2. Run with Loop = True
3. Run with Loop = False
4. See the differences in rendering ( false is totally broken)

### Link to public reproduction project repository

https://github.com/bcaceiro/net9-repros/tree/main/CarouselBug

### Version with bug

9.0.0-rc.1.24453.9

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 14 ( did not test in others)

### Did you find any workaround?

No workaround, just setting Loop="True" which is not the desired effect

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/25192

|Before|After|
|--|--|
|||

### CODE REVIEW

File: src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs [hunk 1]
```diff
@@ -571,46 +571,42 @@ void ClearLayoutListener()
 
 		protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
 		{
-			if (Carousel.Loop)
+			// If the height or width are unbounded and the user is set to
```
Comment (kubaflo) [inline src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs:574]: I only removed the `if (Carousel.Loop)` condition but indentation changes make it harder to spot

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25192.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        }
+
+        public override string Issue => "CarouselView Loop='False' renders incorrectly items";
+
+        [Test]
+        [Category(UITestCategories.CarouselView)]
+        public void CarouselViewShouldRenderCorrectly()
+        {
+            App.WaitForElement("Item1");
+            VerifyScreenshot();
```
[the remark this answers is not in the document]
Comment (kubaflo) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25192.cs:20]: Done

**Comment by jsuarezruiz** [comment]
@rmarinho Thoughts?

**Review by rmarinho — APPROVED** [review]

**Comment by rmarinho** [comment]
Seems this was failing UITests when rebased on main, so I added to revert.

The iOS seems not to render the next item, or the padding is different. 

image

**Comment by kubaflo** [comment]
> Seems this was failing UITests when rebased on main, so I added to revert.
> 
> The iOS seems not to render the next item, or the padding is different.
> 
> image

@rmarinho ah I see.
Item spacing was not working on iOS at the time of adding snapshot. The CarouselView2 for iOS fixed it, so I've updated the iOS screenshot to fix UI test

**Pull request merged**